lnpbp-0021.md

```diff
@@ -56,102 +56,111 @@ Interface specification is the following Contractum code:
 ```haskell
 -- # Defining main data structures
 
--- collectibles are usually scarse, so we limit their max number to 64k
 data ItemsCount :: U32
 
 -- each collectible item is unique and must have an id
-data TokenId :: U16
+data TokenId :: U32
 
 -- Collectibles are owned in fractions of a single item; here the owned
 -- fraction means `1/F`. Ownership of the full item is specified `F=1`;
 -- half of an item as `F=2` etc.
 data OwnedFraction :: U64
 
-data TxOut :: txid [Byte ^ 32], vout U16
-
 -- allocation of a single token or its fraction to some transaction output
 data Allocation :: TokenId, OwnedFraction
 
-data Engraving :: appliedTo TokenId, content Media
+data EngravingData :: 
+    appliedTo TokenId, 
+    content EmbeddedMedia
 
-data Media ::
-    type MimeType,
+data EmbeddedMedia ::
+    type RGBTypes.MediaType,
     data [Byte]
 
 data Attachment ::
-    type MimeType,
+    type RGBTypes.MediaType,
     digest: [U8 ^ 32] -- this can be any type of 32-byte hash, like SHA256(d), BLACKE3 etc
 
-data POR :: -- proof of reserves
-    reserves TxOut,
-    proof [Byte] -- schema-specific proof 
-
-data Token ::
+data TokenData ::
     id TokenId,
-    name [Ascii ^ 1..40],
-    details [Unicode ^ 40..256]?,
-    preview Media?, -- always embedded preview media < 64kb
-    media Attachment?, -- external media which is the main media for the token
+    ticker RGBTypes.Ticker?,
+    name RGBTypes.Name?,
```

Comment:
Why do we need different ticker and name for each token? We already have this data at a global level and specifying extra details for each token should be enough. I'm worried that having possibly different ticker and name for each token may be confusing for the token holder and also could become something not trivial to handle in a user interface